server/repository/src/migrations/README.md

```diff
@@ -17,22 +17,58 @@ During startup server will run [these steps sequentially](mod.rs):
 
-**note** everything after `patch` is considered `pre-release`, and cannot be upgraded further (pre-release version is really undefined, it could be a test branch or a release candidate), but `pre-release` versions can be used to manually test migrations/functionality in production.
-
 Increment [package.json](../../../../package.json) version to the new version and create new migration folder with the version number. Copy template or existing migration, and rename to new version appropriately. 
 
 Add new version mod to [root migrations mode](mod.rs) and add new version to `vec!` of visitors. Add actual migration code and tests, through tests you should be able to check sql syntax and data migration logic without starting server.
 
+## Migration Fragments
+
+We had two issues with migrations:
+
+1 - Migrations were disabled in RC versions, and if we had schema patches, QA team found it hard to constantly re-initialise database and add new test data etc...
+2 - In development, while changing branches, newly added migrations did not run, again requiring manual re-run of those migrations
+
+This is why Migration Fragments were added, via `migration_ragments()` method on Migration trait. 
+
+*With great power comes great responsibility*
+
+There was a deliberate restriction to not allow RC migrations, and to restrict migration to `one time` during develop because it could cause an 'unknown state' of database, as outlined in use cases below. However, most of the time migration patches done in development or in RC testing phase can be executed safely in isolation with Migration Fragments. There are still so many use cases where migrations could cause undefined database state, here are some examples of flexibility of Migration Fragments and gotchas.
+
+1 - We start with adding a [new migration and a new table](https://github.com/msupply-foundation/open-msupply/compare/c95609f818f171bf106e8124c7ee87815d5f996e...4ede6643460459cb9a1bff4bbeea924bdb6c2e54).
+
+2 - [A field is added to that table](https://github.com/msupply-foundation/open-msupply/compare/4ede6643460459cb9a1bff4bbeea924bdb6c2e54...c53a65e19f1894baae7c2568aa8800fb69941a8d) (make sure it has reasonable defaults if it the column is not optional !). You can see that when devs are working in parallel on separate migrations, it should be possible to add consolidate migrations with these method, quite easily (since each fragment should be applied independently).
+
+3 - Modifying a field would require quite a lot of SQLite code, but we [can drop table and re-create it](https://github.com/msupply-foundation/open-msupply/compare/c53a65e19f1894baae7c2568aa8800fb69941a8d...71fbfb75a003f78cb29b678c6d1ade8810b98748).
```

Comment:
This compare is confusing... there's two copies of the `delete_after_rc_add_new_table` file, and the `delete_after_rc_drop_new_table` fragment is called twice??

However, want to chat about this recommendation - I think renaming the existing migration file, then creating a new one with the same name but a different identifier, feels pretty error prone 👀 Could we not just create another fragment called `update_data_to_double`, which is run after `add_new_table`, and in it we drop the table and recreate with the updated columns?